runtime/compiler/env/j9method.cpp

```diff
@@ -6993,7 +6994,8 @@ TR_ResolvedJ9Method::staticAttributes(TR::Compilation * comp, I_32 cpIndex, void
        *unresolvedInCP = isUnresolvedInCP;
 
    bool isColdOrReducedWarm = (comp->getMethodHotness() < warm) || (comp->getMethodHotness() == warm && comp->getOption(TR_NoOptServer));
-   bool doRuntimeResolveForEarlyCompilation = isUnresolvedInCP && isColdOrReducedWarm;
+   bool isMethodHandleThunk = comp->ilGenRequest().details().isMethodHandleThunk() || this->isArchetypeSpecimen();
```

Comment:
There's a similar check in the code below around line 724
```
comp->ilGenRequest().details().isMethodHandleThunk() || // cmvc 195373
```
should it be updated to use this `isMethodHandleThunk` predicate?

Reply:
good catch @DanHeidinga - thoughts @liqunl ?

Reply:
Added